x-pack/plugins/apm/public/components/app/Correlations/LatencyCorrelations.tsx

```diff
@@ -0,0 +1,271 @@
+import {
+  APIReturnType,
+  callApmApi,
+} from '../../../services/rest/createCallApmApi';
+import { px } from '../../../style/variables';
+import { SignificantTermsTable } from './SignificantTermsTable';
+
+type CorrelationsApiResponse = NonNullable<
+  APIReturnType<'/api/apm/correlations/slow_transactions', 'GET'>
+>;
+
+type SignificantTerm = CorrelationsApiResponse['significantTerms'][0];
+
+export function LatencyCorrelations() {
+  const [
+    selectedSignificantTerm,
+    setSelectedSignificantTerm,
+  ] = useState<SignificantTerm | null>(null);
+
+  const { serviceName } = useParams<{ serviceName?: string }>();
+  const { urlParams, uiFilters } = useUrlParams();
+  const { transactionName, transactionType, start, end } = urlParams;
+
+  const { data, status } = useFetcher(() => {
+    if (start && end) {
+      return callApmApi({
+        pathname: '/api/apm/correlations/slow_transactions',
+        params: {
+          query: {
+            serviceName,
+            transactionName,
+            transactionType,
+            start,
+            end,
+            uiFilters: JSON.stringify(uiFilters),
+            durationPercentile: '50',
+            fieldNames:
+              'user.username,user.id,host.ip,user_agent.name,kubernetes.pod.uuid,kubernetes.pod.name,url.domain,container.id,service.node.name',
+          },
+        },
+      });
+    }
+  }, [serviceName, start, end, transactionName, transactionType, uiFilters]);
+
+  return (
+    <>
+      <EuiFlexGroup direction="column">
+        <EuiFlexItem>
+          <EuiFlexGroup direction="row">
+            <EuiFlexItem>
+              <EuiTitle size="s">
+                <h4>Average latency over time</h4>
+              </EuiTitle>
+              <LatencyTimeseriesChart
+                data={data}
+                status={status}
+                selectedSignificantTerm={selectedSignificantTerm}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiTitle size="s">
+                <h4>Latency distribution</h4>
+              </EuiTitle>
+              <LatencyDistributionChart
+                data={data}
+                status={status}
+                selectedSignificantTerm={selectedSignificantTerm}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <SignificantTermsTable
+            significantTerms={data?.significantTerms}
+            status={status}
+            setSelectedSignificantTerm={setSelectedSignificantTerm}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </>
+  );
+}
+
+function LatencyTimeseriesChart({
+  data,
+  selectedSignificantTerm,
+  status,
+}: {
+  data?: CorrelationsApiResponse;
+  selectedSignificantTerm: SignificantTerm | null;
+  status: FETCH_STATUS;
+}) {
+  if (!data) {
+    if (status === FETCH_STATUS.LOADING) {
+      return <EuiLoadingChart size="m" />;
```

Comment:
You should use the `ChartContainer` component that handles failure and loading states so you don't need to update this if we change how those operate.